src/object/srv_obj.c

```diff
@@ -1721,21 +1708,162 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc,
 
 	rc = obj_rw_complete(rpc, ioc, ioh, rc, dth);
 	if (iods_dup != NULL)
-		daos_iod_recx_free(iods_dup, orw->orw_nr);
+		daos_iod_recx_free(iods_dup, iods_nr);
 	return unlikely(rc == -DER_ALREADY) ? 0 : rc;
 }
 
+/* Extract local iods/offs/csums by orw_oid.id_shard from @orw */
 static int
-obj_local_rw(crt_rpc_t *rpc, struct obj_io_context *ioc,
-	     daos_iod_t *split_iods, struct dcs_iod_csums *split_csums,
-	     uint64_t *split_offs, struct dtx_handle *dth)
+obj_get_iods_offs_by_oid(daos_unit_oid_t uoid, struct obj_iod_array *iod_array,
+			 struct daos_oclass_attr *oca, uint64_t dkey_hash,
+			 uint32_t layout_ver, daos_iod_t **iods,
+			 uint64_t **offs, struct dcs_iod_csums **csums, uint32_t *nr)
 {
-	int	rc;
-	int	count = 0;
+	struct obj_shard_iod	*siod;
+	uint32_t		local_tgt;
+	uint32_t		oiod_nr;
+	int			i;
+	int 			idx = 0;
```

Comment:
```suggestion
	int			idx = 0;
```

Reply:
may fix this if you refresh it next time, thx